tests/Doctrine/Tests/Common/Proxy/ProxyGeneratorTest.php

```diff
@@ -229,6 +231,94 @@ public function testClassWithReturnTypesOnProxiedMethods()
         $this->assertEquals(1, substr_count($classCode, 'function returnsParent(): \stdClass'));
     }
 
+    public function testClassWithNullableTypeHintsOnProxiedMethods()
+    {
+        if (PHP_VERSION_ID < 70100) {
+            $this->markTestSkipped('Nullable type hints are only supported in PHP >= 7.1.0.');
+        }
+
+        if (!class_exists('Doctrine\Tests\Common\ProxyProxy\__CG__\NullableTypeHintsClass', false)) {
+            $className = NullableTypeHintsClass::class;
+            $metadata = $this->createClassMetadata($className, ['id']);
+
+            $proxyGenerator = new ProxyGenerator(__DIR__ . '/generated', __NAMESPACE__ . 'Proxy', true);
+            $this->generateAndRequire($proxyGenerator, $metadata);
+        }
+
+        $classCode = file_get_contents(__DIR__ . '/generated/__CG__DoctrineTestsCommonProxyNullableTypeHintsClass.php');
+
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintInt(?int $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintObject(?\stdClass $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintSelf(?\\' . $className . ' $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintWithDefault(?int $param = 123)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintWithDefaultNull(?int $param = NULL)'));
```

Comment:
@trowski Thanks, unfortunately that's not enough either, I used isDefaultValueAvailable which should be the same. See https://github.com/doctrine/common/pull/734/files#diff-ce03ab9b396edcbb313c54234c20e0deR1081

The following:
```php
public function foo(int $param = null)
```
would still generate:
```php
public function foo(?int $param = null)
```
which is not correct (at least in <7.1). And since we don't want any version checks, it's tricky...